utilities/xmlrpcpp/src/XmlRpcClient.cpp

```diff
@@ -312,6 +312,13 @@ XmlRpcClient::generateRequest(const char* methodName, XmlRpcValue const& params)
                   header.length(), body.length());
 
   _request = header + body;
+  // Limit the size of the request to avoid integer overruns
+  if (_request.length() > size_t(__INT_MAX__)) {
+    XmlRpcUtil::error("XmlRpcClient::generateRequest: request length (%u) exceeds maximum allowed size (%u).",
+                      _request.length(), __INT_MAX__);
+    _request.resize(__INT_MAX__);
```

Comment:
I don't understand this truncation; what will happen to the data in `_request` after this? Will it be garbage-collected? Or passed on to something else for further handling? I suspect discarding all the data would be easier / better than keeping two gigabytes of the data.

Reply:
Changed to clear() in 7d51a7aaea0cf022cc82357e03db6340741364d3